apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -194,4 +196,118 @@ defmodule Snitch.Domain.Order.Transitions do
         {:halt, error}
     end)
   end
+
+  # Guard Class to check negative amount is skipped.
+
+  defp calc_payable_amount(order) do
+    packages_total = Package.compute_packages_total(order)
+    total_amount = Money.add!(packages_total, order.total)
+    amount_paid_before = Payment.compute_order_payments(order)
+    Money.sub!(total_amount, amount_paid_before)
+  end
+
+  defp process_payment_chk(context, multi, payment_method, payment, order, amount_to_pay) do
+    params = %{
+      amount: amount_to_pay,
+      order_id: order.id,
+      slug: "check-payment"
+    }
+
+    function = fn _ ->
+      CheckPaymentModel.create(params)
+    end
+
+    struct(context, multi: Multi.run(multi, :checkpayment, function))
+  end
+
+  defp process_payment_ccd(context, multi, payment_method, payment, order, amount_to_pay) do
+    case validate_payment_method(payment) do
+      %{changes: %{card: %{errors: []}}} ->
+        params = %{
+          amount: amount_to_pay,
+          payment_type: "ccd",
+          payment_method_id: payment_method.id
+        }
+
+        function = fn _ ->
+          CardPaymentModel.create("card-payment", order.id, params, %{card: payment})
+        end
+
+        struct(context, multi: Multi.run(multi, :cardpayment, function))
+
+      %{changes: %{card: %{errors: errors}}} ->
+        struct(context, valid?: false, errors: errors)
+    end
+  end
+
+  @doc """
+  Validator for payment method
+
+  Responses
+  {:ok, payment}
+  {:error, errors}
+  """
+
+  defp validate_payment_method(payment) do
```

Comment:
Why is this necessary? If the changeset is invalid, the multi will fail when it runs anyways, pls remove this, and the related complexity in `process_payment_ccd`

Reply:
Just need minor clarification.
If a user `card` payment method id and  enter card details
does the card details enter transition ? or will be controller handles the card details ?

My assumption, that card details are handled in transition.

If card details are handled by controller, we can remove couple of functions.